numpy/random/mtrand/mtrand.pyx

```diff
@@ -123,6 +123,18 @@ cdef extern from "distributions.h":
     long rk_hypergeometric(rk_state *state, long good, long bad, long sample) nogil
     long rk_logseries(rk_state *state, double p) nogil
 
+cdef extern from "mvhypergeom.h":
+    # do not need the GIL, but they do need a lock on the state !!
+
+    int multivar_hypergeom_count(rk_state *state,
+                                 size_t num_colors, long *colors,
```

Comment:
Since this isn't quite in, is it possible to use long long to avoid windows problems?

Reply:
FWIW, I'd recommend using an explicitly-sized integer type like `int64_t` instead of `long long` if we do decide to change anything here. I'm not sure that we ought to change the type in the signature, though. I'm guessing that inputs above `2**31` are unlikely and probably not going to work well with the algorithm regardless.

Internal variables holding intermediates might be worth upgrading to `int64_t`. I might recommend doing this kind of thing as a uniform change across all of the distributions, all at once, though.

Reply:
On Windows `long` can be a pain since `int64` has become so common via pandas. I agree that explicit `int64` should be the default type for integer generators.

Reply:
The integers in, for example, `distributions.c` are all `long`, so I'm basically following the existing pattern.  If I had to justify it, I'd say that using `long` matches the default type of integer arrays in numpy (see, for example, Nathaniel's recent comment here: https://github.com/numpy/numpy/issues/9464#issuecomment-318185121).

Robert's suggestion to eventually update the types all at once sounds good.

Reply:
That was indeed the original motivation for using `long` in all of the other distributions' signatures.

Reply:
I agree that this isn't the place to make those changes.  I do think accepting `int64` for integer datatypes is generally a good idea.